platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/tasks/CachingTaskDependencyResolveContext.java

```diff
@@ -58,14 +59,27 @@ public CachingTaskDependencyResolveContext(Collection<? extends WorkDependencyRe
         this.walker = new CachingDirectedGraphWalker<>(new TaskGraphImpl(workResolvers));
     }
 
+    @Override
+    public String getName() {
```

Comment:
I'm not really sure it seems right for a `TaskDependencyResolveContext` to have a name like this. In practice, whether or not there is a task used as context in the `getDependencies` method, we are still resolving task dependencies. 

I think we should remove the `getName` method and continue to hard code "task dependencies" in `ResolutionBackedFileCollection.java` like we were before

Reply:
I strongly disagree: If it's a task dependency, then which task can they change to fix the error?

I don't see the issue with naming a context, either.  The type hierarchy of `Named` is extensive, and there seem to be much bigger "stretches" of the concept already present.

Reply:
I think we should get the execution team's opinion on this. They're the ones who own this stuff here

Even if we think these two separate instances of task dependency resolve contexts should have different error messages, I don't think we should express that as a `getName` field. Names are traditionally used to identify specific instances of a thing. I wouldn't expect (within one project) to have two different configurations with the same name. I would expect the method to be called `getType`. 

In fact, the method we pass this value into, `resolutionHost#mapFailure` names this parameter `type`

Reply:
There is also the concept of "Build Dependenices" and I'm not sure how that relates here

Reply:
Would `Describable` be more appropriate?  `Named` and `getName` imply what you'd call the domain object, but this is something we need that's human readable.

IOW, if `TaskDependencyResolveContext` had a name, it might be something complicated and unique, but its display name might be something more generic.

Reply:
`Describable` does seem like a better choice after reading the javadoc.  I'll make this change.

Does this address your concerns here, Justin?

Reply:
Somewhat, though I still think we should hardcode the value. I think we should pick a value here that covers both situations. 

There is a name for the type of thing we're resolving (and failing to resolve) in this case. Its either called "task dependencies", "build dependencies", or just "dependencies", and I think the @gradle/bt-execution team should decide

Reply:
I still disagree: this type behaves differently based on whether or not the `task` field is `null`.  The description should also vary with this.

A message should definitely not say `task dependencies` if the issue would still be present in the build if all tasks were removed.  I'm open to it saying something more unique than just "dependencies", but at a minimum we need to not mislead the reader about where the problem lies.